UI update

lognece opened the pull request.

lognece committed:
Attempt to implement a custom function to update the constant from the ui

lognece committed:
Fixed the update_constant in custom_backend to dynamically update the constant from the ui

lognece committed:
- Custom backend function update_constant() now works as it should.
- updated READMEs
- integrated update_constant() in the new UI

lognece committed:
Changed the Repo format and made minor changes to the ui appearance

lognece committed:
Changed the Repo format and made minor changes to the ui appearance (added missing files)

lognece committed:
Changed UI:
- Added credit information (Total + sum for each semester + for each module)
- Added buttons to the menu bar
- "Download Study Plan" now opens a modal prompting the user to enter the file name
- Added a module information modal

Custom Backend:
- Added a custom download method that formats the output to be user-friendly and resemble a study plan

lognece committed:
Added description of the custom_download method to the README

lognece committed:
added screenshots

lognece committed:
added screenshots to the README

lognece committed:
added side bar with subsets of modules and calculations of remaning credits for each subset

lognece committed:
added responsive layout, progress bar

lognece committed:
- added a new container with modules with checkboxes for the custom selection of modules in the drop-down menu
- Improved the responsive layout
- added a modified version of the custom backend (adjusted to the new version of clinguin)
- refactored the code (improved the names and comments)

susuhahnml reviewed (changes requested):
So I think there is something strange. If you dont add a module to the prefered ones then it is not showed in the plan
for instance if you un-check the mcs
you just dont see it in the plan
perhaps those that have to be included should be disabled
so that they remain always ticked
what do you think?

lognece committed:
- ui_extra.lp has additional left panel for module preference
- ui_main.lp - added modals to the modules in the plan
- cogsys_info.lp - detailed information for the study program parsed from the file "study_programs_info.xml"
- added encoding for the examinations from study-regulations repo

lognece committed:
updated cogsys _info.lp: added explanations

lognece committed:
added modal to the module

susuhahnml committed:
Changes sus

lognece committed:
preference + modal improvement content

susuhahnml committed:
Improved UI with preferences

lognece committed:
- separated huge ui file into parts: layout, menu bar, modals, modules, semesters, subsets and included them all in ui_main.lp
- added requirements.txt
- added instructions how to install and run the program
- in ui: added preference-icons to the subset buttons + made some changes to the modal

lognece committed:
Changes in the Modal

susuhahnml committed:
Update ideas for UI and functions

lognece committed:
Small adjustments to the layout

lognece committed:
added exams

lognece committed:
- use clinguin 2.3.1
- added sidebar and put subsets inside
- added drag and drop functionality
- added collapse elements for subsets

lognece committed:
fixed preferences + added name of the program

lognece committed:
upgraded clinguin, updated README, combined encodings and instances in main files

lognece committed:
updated instructions
